src/solvers/nonlinear_solid_operators.cpp

```diff
@@ -28,24 +28,24 @@ mfem::Operator &NonlinearSolidQuasiStaticOperator::GetGradient(const mfem::Vecto
 // destructor
 NonlinearSolidQuasiStaticOperator::~NonlinearSolidQuasiStaticOperator() {}
 
-NonlinearSolidDynamicOperator::NonlinearSolidDynamicOperator(std::shared_ptr<mfem::ParNonlinearForm> H_form,
-                                                             std::shared_ptr<mfem::ParBilinearForm>  S_form,
-                                                             std::shared_ptr<mfem::ParBilinearForm>  M_form,
-                                                             const mfem::Array<int> &                ess_tdof_list,
-                                                             mfem::NewtonSolver &                    newton_solver,
-                                                             const LinearSolverParameters &          lin_params)
+NonlinearSolidDynamicOperator::NonlinearSolidDynamicOperator(
+    std::shared_ptr<mfem::ParNonlinearForm> H_form, std::shared_ptr<mfem::ParBilinearForm> S_form,
+    std::shared_ptr<mfem::ParBilinearForm> M_form, const std::vector<std::shared_ptr<BoundaryConditionData> > &ess_bdr,
+    mfem::NewtonSolver &newton_solver, const LinearSolverParameters &lin_params)
     : mfem::TimeDependentOperator(M_form->ParFESpace()->TrueVSize() * 2),
       m_M_form(M_form),
       m_S_form(S_form),
       m_H_form(H_form),
       m_newton_solver(newton_solver),
-      m_ess_tdof_list(ess_tdof_list),
+      m_ess_bdr(ess_bdr),
       m_lin_params(lin_params),
       m_z(height / 2)
 {
   // Assemble the mass matrix and eliminate the fixed DOFs
   m_M_mat.reset(m_M_form->ParallelAssemble());
-  auto Me = std::unique_ptr<mfem::HypreParMatrix>(m_M_mat->EliminateRowsCols(m_ess_tdof_list));
+  for (auto &ess_bc_data : m_ess_bdr) {
+    auto Me = std::unique_ptr<mfem::HypreParMatrix>(m_M_mat->EliminateRowsCols(ess_bc_data->true_dofs));
+  }
```

Comment:
is `Me` used for anything?

Reply:
Not right now. It's the eliminated rows of the mass matrix, which is needed if you're enforcing changing displacement boundary conditions. I have this tagged as a feature to add, but haven't gotten around to it yet.